### PR TITLE
refactor toastr, add test to footer toggling

### PR DESCRIPTION
**Pull request by goldylucks**

### CODE REVIEW

File: client/actions/logger.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
 
 import { createAction } from 'redux-actions';
+import { identity } from '../util';
+import { metaToastr } from '../util/toastr';
 
-export const updateLogging = createAction('update logging');
+export const updateLogging = createAction('update logging', identity,
+  flag => metaToastr(`logging updated to ${flag}`)
```
Comment (tomusdrw) [inline client/actions/logger.js]: IMHO syntax:
```js
createAction('update logging', identity, withToastr(flag => 'loggin updated to ${flag}'))
```
would be much more readable.

File: client/middleware/toastr.spec.js [hunk 2]
```diff
@@ -19,195 +20,157 @@ describe('MIDDLEWARE: TOASTR', () => {
...
-      })).to.be.true;
+      expect(next.calledWith(addToast({
+        msg, type, toastNo
+      }))).to.be.true;
 
-      setTimeout(() => {
-        expect(cut.removeToast.calledWith(next, toastNo)).to.be.true;
-        done();
-      }, time);
+      // expect(cut.setTimeoutFor.calledWith(toastNo, next));
```
Comment (tomusdrw) [inline client/middleware/toastr.spec.js]: to remove?

File: client/actions/logger.js (hunk 1 above)
Comment (goldylucks) [inline client/actions/logger.js]: agree

File: client/middleware/toastr.spec.js (hunk 2 above)
Comment (goldylucks) [inline client/middleware/toastr.spec.js]: forgot to added back

**Pull request merged**